src/scores/continuous/consistent_impl.py

```diff
@@ -0,0 +1,262 @@
+"""
+Implementation of scoring functions that are consistent for
+single-valued forecasts targeting quantiles, expectiles or Huber functionals.
+"""
+
+from typing import Callable, Optional
+
+import xarray as xr
+
+from scores.functions import apply_weights
+from scores.typing import FlexibleDimensionTypes
+from scores.utils import gather_dimensions
+
+
+def consistent_expectile_score(
+    fcst: xr.DataArray,
+    obs: xr.DataArray,
+    *,  # Force keywords arguments to be keyword-only
+    alpha: float,
+    phi: Callable[[xr.DataArray], xr.DataArray],
+    phi_prime: Callable[[xr.DataArray], xr.DataArray],
+    reduce_dims: Optional[FlexibleDimensionTypes] = None,
+    preserve_dims: Optional[FlexibleDimensionTypes] = None,
+    weights: Optional[xr.DataArray] = None,
+) -> xr.DataArray:
+    """
+    Returns the score using a scoring function that is consistent for the
+    alpha-expectile functional, based on a supplied convex function phi.
+    See Geniting (2011), or Equation (10) from Taggart (2021).
+
+    Args:
+        fcst: array of forecast values.
+        obs: array of corresponding observation values.
+        alpha: expectile level. Must be strictly between 0 and 1.
+        phi: a convex function on the real numbers, accepting a single array like argument.
```

Comment:
While there is a requirement on phi to be a convex function ... , its purpose is to map the error values as required for consistency. Is it more useful to explain the purpose of phi than the type information for phi? To what extent do we think the API docs will be read by people with the relevant maths background vs people who might appreciate a more functional/purposeful explanation of the arguments? I'm just asking the question. In this case, I think anyone who doesn't understand the function in a fair amount of detail probably shouldn't be using it. Perhaps a link to a tutorial in the doctoring would be a useful thing.

Reply:
I think `consistent_impl` will only be used by more advanced users who a comfortable with the maths. When we add the threshold weighted scores, it will provide users with a much easier set of scores to work with (since they call these functions). 

I will however come back to this and see if there is a nice way for the docstring to be improved. Users of this really need to read through the papers cited (or at least work through the tutorial which provides the minimum amount of knowledge to understand what's going on).

Reply:
Adding a link to the tutorial seems the simplest thing in that case, and then make sure the tutorial has the references in it.

Reply:
I have now added a link to the tutorial